### ISSUE TITLE
fails to compile on x86 / 32-bit mode on windows

### ISSUE THREAD

**Issue by janwilmans** [copperspice/copperspice#89]
r10 is used, so probably makes sense that it fails in 32 bit as that is a 64-bit register.

proposal: add a 32-bit version _or_ if we do not support 32-bit make it fail with proper error message at the beginning... 

```
Microsoft (R) Macro Assembler Version 14.16.27026.1
Copyright (C) Microsoft Corporation.  All rights reserved.

 Assembling: ..\src\3rdparty\webkit\Source\WebCore\plugins\win\PaintHooks.asm
..\src\3rdparty\webkit\Source\WebCore\plugins\win\PaintHooks.asm(27) : error A2006:undefined symbol : r10
..\src\3rdparty\webkit\Source\WebCore\plugins\win\PaintHooks.asm(42) : error A2006:undefined symbol : r10
[14/239] Building CXX object src\xmlpatterns\CMakeFiles\CsXmlPatterns1.5.dir\functions\qpatternmatchingfns.cpp.obj
ninja: build stopped: subcommand failed.
```

This is would could make it fail fast: 
```
if (NOT "${CMAKE_SIZEOF_VOID_P}" STREQUAL "8")
    message(FATAL_ERROR "Only compiling on 64bit targets is supported")
endif()

```

**Comment by agserm** [comment]
It looks like this file should only be part of the build if you are on MSVC *and* you are building a 64-bit version. Right now the cmake build just tests for MSVC.

**Comment by janwilmans** [comment]
Shall I try to create a fix?

**Comment by agserm** [comment]
Yes, please do see if you can put together a fix. It should be as simple as changing the existing test in the cmake file from "MSVC" to "MSVC and 64-bit."

**Comment by agserm** [comment]
The fix for this has been merged in 2f64796. Thanks for the contribution.
